migration/transitioning.rst

```diff
@@ -0,0 +1,126 @@
+
+Migration tactics differ between the two main Citus use cases of multi-tenant applications and real-time analytics. The former is requires fewer data model changes so we'll begin there.
+
+Multi-tenant Data Model
+=======================
+
+Citus is well suited to hosting B2B multi-tenant application data. In this model application tenants share a Citus cluster and a schema and Citus keeps each tenant's data private. Each tenant's table data is stored in its own shard determined by a configurable tenant id column. Citus pushes queries down to run directly on the relevant tenant shard in the cluster, spreading out the computation. Once queries are routed this way they can be executed without concern for the rest of the cluster. These queries can use the full features of SQL, including joins and transactions, without running into the inherent limitations of a distributed system.
+
+This section will explore how to model for the multi-tenant scenario, including necessary adjustments to the schema and queries.
+
+Schema Migration
+----------------
+
+Migrating from a standalone database instance to a sharded multi-tenant system requires identifying and modifying three types of tables which we may term *per-tenant*, *reference*, and *global*. The distinction hinges on whether the tables have (or reference) a column serving as tenant id. The concept of tenant id depends on the application and who exactly are considered its tenants.
+
+Consider an example multi-tenant application similar to Etsy or Shopify where each tenant is a store. Here's portion of a simplified schema:
+
+.. code::
+
+  admins
+    email
+    password
+
+  stores
+    id
+    owner_email
+    owner_password
+    name
+    url
+
+  products
+    id
+    store_id
+    name
+    description
+    category
+
+  purchases
+    id
+    store_id
+    product_id
+    customer_id
+    region_id
+    price
+    purchased_at
+
+  regions
+    id
+    sales_tax
+
+We'll classify the types of tables and learn how to migrate them to Citus.
+
+Tenant-Specific Tables
+^^^^^^^^^^^^^^^^^^^^^^
+
+In our example each store is a natural tenant. This is because storefronts benefit from dedicated processing power for their customer data, and stores should not be able to access each other's sales or inventory. The tenant id is in this case the store id. Notice that some tables already have a store id as primary or foreign key. Tables such as these which can join on the tenant id are called *tenant-specific* or *per-tenant* tables. In this example they are stores, products and purchases.
+
+Global and Reference Tables
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The tenant-specific tables do not require modification for use in Citus. During migration you would distribute them through the Citus cluster, sharding by tenant id. The other tables — admins and regions — do require modification.
```

Comment:
The definition I was going with was that per-tenant tables are ones that already have the tenant id in the db to be migrated, whereas reference tables do not yet have it but will need it.
> Notice that some tables already have a store id as primary or foreign key. Tables such as these which can join on the tenant id are called *tenant-specific* or *per-tenant* tables.

In the example schema for instance we included the tenant id in stores, products and purchases. However in reality perhaps the schema to be migrated may not even have the store table yet because it was designed for only one store. The store table would need to be created and the store_id added to products and purchases.

Maybe we should remove stores from the initial schema to make this migration example more substantial.